src/overlays/actors/ovl_En_Siofuki/z_en_siofuki.c

```diff
@@ -21,27 +30,286 @@ const ActorInit En_Siofuki_InitVars = {
     (ActorFunc)EnSiofuki_Update,
     (ActorFunc)EnSiofuki_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/EnSiofuki_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/EnSiofuki_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+extern Gfx D_06000B70[];
+extern UNK_TYPE D_06000D78;
+
+void EnSiofuki_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSiofuki* this = THIS;
+    s32 param;
+    s32 localC = 0;
+    s32 pad;
+
+    if ((thisx->room == 10) && Flags_GetSwitch(globalCtx, 0x1E)) {
+        Actor_Kill(thisx);
+        return;
+    }
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, DPM_PLAYER);
+    DynaPolyInfo_Alloc(&D_06000D78, &localC);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, localC);
```

Comment:
```suggestion
    ColHeader* colHeader = NULL;
    s32 pad;

    if ((thisx->room == 10) && Flags_GetSwitch(globalCtx, 0x1E)) {
        Actor_Kill(thisx);
        return;
    }

    Actor_ProcessInitChain(thisx, sInitChain);
    DynaPolyInfo_SetActorMove(&this->dyna, DPM_PLAYER);
    DynaPolyInfo_Alloc(&D_06000D78, &colHeader);
    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
    this->sfxFlags |= 1;
```
I believe this is the current standard for dynapoly actors.